iree_tests/benchmarks/benchmark_sdxl_rocm.py

```diff
@@ -1,36 +1,132 @@
+# Copyright 2024 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
 import os
 from collections import namedtuple
+import logging
+from typing import Sequence
+import subprocess
 
 benchmark_dir = os.path.dirname(os.path.realpath(__file__))
+iree_root = os.path.dirname(benchmark_dir)
+prompt_encoder_dir = f"{iree_root}/pytorch/models/sdxl-prompt-encoder-tank"
+scheduled_unet_dir = f"{iree_root}/pytorch/models/sdxl-scheduled-unet-3-tank"
+vae_decode_dir = f"{iree_root}/pytorch/models/sdxl-vae-decode-tank"
 
-def launchCommand(scriptcommand):
-    print("Launching:", scriptcommand, "[ Proc:", os.getpid(), "]")
-    try:
-        ret = os.system(scriptcommand)
-        return ret
-    except OSError as errormsg:
-        print(
-            "Invoking ",
-            scriptcommand,
-            " failed:",
-            errormsg,
-        )
-        return 1
+def run_iree_command(args: Sequence[str] = ()):
+    command = "Exec:", " ".join(args)
+    logging.getLogger().info(command)
+    proc = subprocess.run(args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, check=False)
+    stdout_v, stderr_v, = proc.stdout, proc.stderr
+    return_code = proc.returncode
+    if return_code == 0:
+        return 0, proc.stdout
+    logging.getLogger().info(f"Command failed with error: {proc.stderr}")
+    return 1, proc.stdout
 
 def run_sdxl_rocm_benchmark():
-    scriptcommand = (
-        "bash "
-        + f"{benchmark_dir}/benchmark_sdxl_rocm.sh"
-    )
-    return launchCommand(scriptcommand)
+    exec_args = [
+        "iree-compile",
+        f"{benchmark_dir}/sdxl_pipeline_bench_f16.mlir",
+        "--iree-hal-target-backends=rocm",
+        "--iree-rocm-target-chip=gfx90a",
+        "--iree-global-opt-propagate-transposes=true",
+        "--iree-codegen-llvmgpu-use-vector-distribution",
+        "--iree-codegen-gpu-native-math-precision=true",
+        "--iree-rocm-waves-per-eu=2",
+        "--iree-opt-outer-dim-concat=true",
+        "--iree-llvmgpu-enable-prefetch",
+        "-o",
+        f"{benchmark_dir}/sdxl_full_pipeline_fp16_rocm.vmfb",
+    ]
+    # iree compile command for full sdxl pipeline
+    ret_value, stdout = run_iree_command(exec_args)
+    if ret_value == 1:
+        return 1, stdout
+    exec_args = [
+        "iree-benchmark-module",
+        "--device=hip://6",
```

Comment:
Similar question - let the device ID come from a flag or environment variable instead of hardcoding?